### ISSUE TITLE
Cannot create sound from byte array

### ISSUE THREAD

**Issue by kr8gz** [lebedec/libfmod#11]
I am currently trying to create a sound from an in-memory byte array in OGG format, using `System::create_sound` with `Mode::OPENMEMORY` to load the array. The first parameter of the function takes a `&str` though, which means that it has to be valid UTF-8 and also not contain any null bytes in it.
I have taken a look at the source code for this crate, and identified what should be changed in order for this to work:
https://github.com/lebedec/libfmod-gen/blob/main/src/generators/lib.rs#L714-L715
I don't know much about Rust FFI or C types so any help is appreciated :)

**Comment by lebedec** [comment]
Hi! This is probably an incorrect type determination. Instead of `&str` there should be `[u8]`. I'll see how can I fix it. If you can provide OGG file and exact code sample, it will help a lot.

**Comment by kr8gz** [comment]
Here is an example sound: https://geometrydashfiles.b-cdn.net/sfx/s4451.ogg
Currently, the bytes are fetched using the reqwest crate and stored in an in-memory cache. As for the code to play this byte array, it is nonexistent since it can't be passed to the `create_sound` function and I inserted some placeholder code instead

**Comment by lebedec** [comment]
The problem is more complicated than I thought. FMOD function `System::create_sound` has 3 overloads of first argument depends on usage scenario. It can be file name, memory pointer or byte buffer. Unfortunately in C/C++ development world is regular thing =)

I need more time to figure out how to implement it correctly.

**Comment by lebedec** [comment]
Refs to FMOD documentation: 
* https://www.fmod.com/docs/2.02/api/core-api-system.html#system_createsound
* https://www.fmod.com/docs/2.02/api/core-api-common.html#fmod_mode

**Comment by lebedec** [comment]
It takes a lot of time to implement (maybe weeks).

Here workaround, you can use FFI interface to play OGG from byte array right now:

```rust
#[test]
fn test_playing_sound_from_data() -> Result<(), Error> {
    let system = System::create()?;
    system.init(512, FMOD_INIT_NORMAL, None)?;
    let data = fs::read("./tests/data/Assets/1.ogg").unwrap();
    // let sound = system.create_sound("./tests/data/Assets/1.ogg", FMOD_DEFAULT, None)?;
    let sound = unsafe {
        let mut sound = null_mut();
        let mut exinfo: FMOD_CREATESOUNDEXINFO = std::mem::zeroed();
        exinfo.cbsize = std::mem::size_of::<FMOD_CREATESOUNDEXINFO>() as i32;
        exinfo.length = data.len() as u32;
        match ffi::FMOD_System_CreateSound(
            system.as_mut_ptr(),
            data.as_ptr() as *const _,
            FMOD_OPENMEMORY,
            &mut exinfo as *mut _,
            &mut sound,
        ) {
            ffi::FMOD_OK => Ok(Sound::from(sound)),
            error => Err(Error::Fmod {
                function: "FMOD_System_CreateSound".to_string(),
                code: error,
                message: map_fmod_error(error).to_string(),
            }),
        }
    };
    let channel = system.play_sound(sound?, None, false)?;
    while channel.is_playing()? {
        // do something else
    }
    system.release()
}
```

**Comment by kr8gz** [comment]
Thank you for the workaround and taking the time to look into this!

**Comment by lebedec** [comment]
Hi there! Fixed in the new version `libfmod = "2.206.4"`. Now you can use:
```rust
#[test]
fn test_playing_sound_from_data() -> Result<(), Error> {
    let system = System::create()?;
    system.init(512, Init::NORMAL, None)?;
    let data = fs::read("./tests/data/Assets/1.ogg").unwrap();
    
    let mut info = CreateSoundexInfo::default();
    info.length = data.len() as u32;
    let sound = system.create_sound_from(&data, Mode::OPENMEMORY, info)?;

    let channel = system.play_sound(sound, None, false)?;
    while channel.is_playing()? {
        // do something else
    }
    system.release()
}
```
